cmd/snap-preseed/reset.go

```diff
@@ -0,0 +1,105 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2020 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package main
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/snapcore/snapd/dirs"
+)
+
+func resetPreseededChroot(preseedChroot string) error {
+	// globs that yield individual files
+	globs := []string{
+		dirs.SnapStateFile,
+		dirs.SnapSystemKeyFile,
+		filepath.Join(dirs.SnapBlobDir, "*.snap"),
+		filepath.Join(dirs.SnapUdevRulesDir, "*-snap.*.rules"),
+		filepath.Join(dirs.SnapBusPolicyDir, "snap.*.*.conf"),
+		filepath.Join(dirs.SnapServicesDir, "snap.*.service"),
+		filepath.Join(dirs.SnapServicesDir, "snap.*.timer"),
+		filepath.Join(dirs.SnapServicesDir, "snap.*.socket"),
+		filepath.Join(dirs.SnapServicesDir, "snap-*.mount"),
+		filepath.Join(dirs.SnapServicesDir, "multi-user.target.wants", "snap-*.mount"),
+	}
+
+	for _, gl := range globs {
+		matches, err := filepath.Glob(filepath.Join(preseedChroot, gl))
+		if err != nil {
+			// the only possible error from Glob() is ErrBadPattern
+			return err
+		}
+		for _, path := range matches {
+			if err := os.Remove(path); err != nil {
+				// report the error and carry on
```

Comment:
Curious why we not error here, it seems slightly dangerous to just carry on?

Reply:
Yeah, I think you're right. My original thinking was that reset should never fail, but realistically, it's unlikely for this to fail but if it fails, it may indeed be better to error out. Will change it. Thanks.

Reply:
Done.